client/components/AvailabilityGrid/AvailabilityGrid.js

```diff
@@ -516,17 +627,34 @@ class AvailabilityGrid extends React.Component {
   }
 }
 
+AvailabilityGrid.defaultProps = {
+  heatmap: false,
+  user: {},
+  event: {},
+  myAvailability: [],
+  participants: [],
+  submitAvail: () => { console.log('submitAvail func not passed in!'); },
+  editAvail: () => { console.log('ediAvail func not passed in!'); },
+  closeGrid: () => { console.log('closeGrid func not passed in!'); },
+};
+
 AvailabilityGrid.propTypes = {
-  dates: React.PropTypes.array.isRequired,
+  dates: React.PropTypes.arrayOf(React.PropTypes.object).isRequired,
   heatmap: React.PropTypes.bool,
-  user: React.PropTypes.object,
-  availability: React.PropTypes.array,
+  user: React.PropTypes.shape({
```

Comment:
can you change the name to curUser ? we use at all app curUser for the actual user... and using maybe cause some doubts  if is the current user or another user"guest"